scripts/build/prepare_llext_exptab.py

```diff
@@ -0,0 +1,365 @@
+        self.offset = self.section['sh_offset']
+
+class LLEXTExptabManipulator():
+    """Class used to wrap the LLEXT export table manipulation."""
+    def __init__(self, elf_fd, exptab_file_offset, lcs_struct, exports_count):
+        self.fd = elf_fd
+        self.exports_count = exports_count
+        self.base_offset = exptab_file_offset
+        self.lcs_struct = lcs_struct
+
+    def _seek_to_sym(self, index):
+        self.fd.seek(self.base_offset + index * self.lcs_struct.size)
+
+    def __getitem__(self, index):
+        if not isinstance(index, int):
+            raise TypeError(f"invalid type {type(index)} for index")
+
+        if index >= self.exports_count:
+            raise IndexError(f"index {index} is out of bounds (max {self.exports_count})")
+
+        self._seek_to_sym(index)
+        return self.lcs_struct.unpack(self.fd.read(self.lcs_struct.size))
+
+    def __setitem__(self, index, item):
+        if not isinstance(index, int):
+            raise TypeError(f"invalid type {type(index)} for index")
+
+        if index >= self.exports_count:
+            raise IndexError(f"index {index} is out of bounds (max {self.exports_count})")
+
+        (addr_or_slid, sym_addr) = item
+
+        self._seek_to_sym(index)
+        self.fd.write(self.lcs_struct.pack(addr_or_slid, sym_addr))
+
+class ZephyrElfExptabPreparator():
+    """Prepares the LLEXT export table of a Zephyr ELF.
+
+    Attributes:
+        elf_path: path to the Zephyr ELF to prepare
+        log: a logging.Logger object
+        slid_listing_path: path to the file where SLID listing should be saved
+    """
+    def __init__(self, elf_path: str, log: logging.Logger, slid_listing_path: str | None):
+        self.elf_path = elf_path
+        self.elf_fd = open(self.elf_path, 'rb+')
+        self.elf = ELFFile(self.elf_fd)
+        self.log = log
+
+        # Lazy-open the SLID listing file to ensure it is only created when necessary
+        self.slid_listing_path = slid_listing_path
+        self.slid_listing_fd = None
+
+    def _prepare_exptab_for_slid_linking(self):
+        """
+        IMPLEMENTATION NOTES:
+          In the linker script, we declare the export names table
+          as starting at address 0. Thanks to this, all "pointers"
+          to that section are equal to the offset inside the section.
+          Also note that symbol names are always NUL-terminated.
+
+          The export table is sorted by SLID in ASCENDING order.
+        """
+        def read_symbol_name(name_ptr):
+            raw_name = b''
+            self.elf_fd.seek(self.expstrtab_section.offset + name_ptr)
+
+            c = self.elf_fd.read(1)
+            while c != b'\0':
+                raw_name += c
+                c = self.elf_fd.read(1)
+
+            return raw_name.decode("utf-8")
+
+        #1) Load the export table
+        exports_list = []
+        for (name_ptr, export_address) in self.exptab_manipulator: #TODO: this works?
```

Comment:
Slightly concerning TODO here.... =D

Reply:
Indeed :D
Leftover comment, script seems to work fine, will be removing it.